### ISSUE TITLE
Rething error reporting

### ISSUE THREAD

**Issue by alnsn** [arcapos/luaunix#6]
unix.errno is almost useless in Lua. For instance, if GC runs between these two lines:

local d = unix.opendir()
if not d then error(unix.errno()) end

it will likely call some C function that resets errno.

Lua modules usually return error messages in the second return value if they fail:

return nil, "No such file or directory"

Some libraries return a special object instead of a string.

**Comment by mbalmer** [comment]
Yes, I agree.  errno is a C thing, and luaunix should feel "Lua-ish".
